applications/main/clock/clock_settings_app.c

```diff
@@ -0,0 +1,102 @@
+#include <furi.h>
+#include <gui/modules/variable_item_list.h>
+#include <gui/view_dispatcher.h>
+#include <lib/toolbox/value_index.h>
+#include "clock_settings.h"
+
+#define TAG "Clock"
+
+typedef struct {
+    ClockSettings clock_settings;
+    Gui* gui;
+    ViewDispatcher* view_dispatcher;
+    VariableItemList* variable_item_list;
+} ClockAppSettings;
+
+static uint32_t clock_app_settings_exit(void* context) {
+    UNUSED(context);
+    return VIEW_NONE;
+}
+
+#define TIME_FORMAT_COUNT 2
+const char* const time_format_text[TIME_FORMAT_COUNT] = {
+    "12h",
+    "24h",
+};
+
+const uint32_t time_format_value[TIME_FORMAT_COUNT] = {H12, H24};
+
+#define DATE_FORMAT_COUNT 2
+const char* const date_format_text[DATE_FORMAT_COUNT] = {
+    "mm-dd", // ISO 8601
+    "dd-mm", // RFC 5322
+};
+
+const uint32_t date_format_value[DATE_FORMAT_COUNT] = {Iso, Rfc};
+
+static void time_format_changed(VariableItem* item) {
+    ClockAppSettings* app = variable_item_get_context(item);
+    uint8_t index = variable_item_get_current_value_index(item);
+    variable_item_set_current_value_text(item, time_format_text[index]);
+    app->clock_settings.time_format = time_format_value[index];
+}
+
+static void date_format_changed(VariableItem* item) {
+    ClockAppSettings* app = variable_item_get_context(item);
+    uint8_t index = variable_item_get_current_value_index(item);
+    variable_item_set_current_value_text(item, date_format_text[index]);
+    app->clock_settings.date_format = date_format_value[index];
+}
+
+static ClockAppSettings* alloc_settings() {
```

Comment:
Always prefix methods with application name in applications compiled into firmware.

Reply:
Yes, static keyword makes it local, but debugging application still becomes hell